### ISSUE TITLE
clip gradients error ( p34 )

### ISSUE THREAD

**Issue by ONNONS** [ultralytics/yolov5#9834]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello bro I am testing with yolov5-p34 yaml but the following error occurs 

I tried yolov5 nano model and small model, but the error is the same What is the problem?

rain.py:323: FutureWarning: Non-finite norm encountered in torch.nn.utils.clip_grad_norm_; continuing anyway. Note that the default behavior will change in a future release to error out if a non-finite total norm is encountered. At that point, setting error_if_nonfinite=false will be required to retain the old behavior.
  torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm=10.0)  # clip gradients

**Comment by glenn-jocher** [comment]
Seems like a warning rather than an error.

**Comment by ONNONS** [comment]
> With that warning, box loss and obj loss are marked as nan and cls loss is marked as zero, so it is not learned properly

**Comment by glenn-jocher** [comment]
@ONNONS got it. If you are using default hyps then you might want to lower your learning rate or increase your warmup etc.

**Comment by ONNONS** [comment]
> @ONNONS got it. If you are using default hyps then you might want to lower your learning rate or increase your warmup etc.

Changing the hyps will also generate a warning and cause the training to fail. but yolov5n6 learns without warning yolov5-p34 How to use yaml?

**Comment by glenn-jocher** [comment]
@ONNONS got it. You can start training from any yaml with the --cfg argument, i.e.:
```
python train.py --weights yolov5s.pt --cfg yolov5-p34.yaml
```

**Comment by ONNONS** [comment]
> @ONNONS got it. You can start training from any yaml with the --cfg argument, i.e.:
> 
> ```
> python train.py --weights yolov5s.pt --cfg yolov5-p34.yaml
> ```
Warnings continue to occur. Do you have p34 working?

torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm=10.0) # clip gradients

**Comment by glenn-jocher** [comment]
@ONNONS all the models train correctly on the default datasets, so if there are gradient issues you might want to inspect your dataset to verify your labels.
